services/ui-src/src/components/overlays/EntityDetailsDashboardOverlay.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { MouseEventHandler, useContext, useEffect } from "react";
+// components
+import { Box, Button, Flex, Image, Spinner } from "@chakra-ui/react";
+import { Form, ReportPageIntro } from "components";
+// types
+import { EntityShape, EntityType, FormJson } from "types";
+// assets
+import arrowLeftBlue from "assets/icons/icon_arrow_left_blue.png";
+// verbiage
+import overlayVerbiage from "../../verbiage/pages/overlays";
+import { EntityContext } from "components/reports/EntityProvider";
+
+export const EntityDetailsDashboardOverlay = ({
+  closeEntityDetailsOverlay,
+  entityType,
+  entities,
+  form,
+  onSubmit,
+  selectedEntity,
+  disabled,
+  submitting,
+  validateOnRender,
+}: Props) => {
+  // Entity Provider Setup
+  const { setEntities, setSelectedEntity, setEntityType } =
+    useContext(EntityContext);
+
+  useEffect(() => {
+    setSelectedEntity(selectedEntity);
+    setEntityType(entityType);
+    setEntities(entities);
+    return () => {
+      setEntities([]);
+      setSelectedEntity(undefined);
+    };
+  }, [entityType, selectedEntity]);
```

Comment:
kind of curious how this works 🤔 I haven't seen a return in useEffect before

Reply:
i'm guessing this might have been copy and pasted from an old file, this is a lot of placeholder code

Reply:
Yeah Casie's right! I switched the name of this file and nothing else - so I'm not actually sure what's going on in this file tbh

Reply:
Yea I think you can remove the return and the setEntities & setSelectedEntity should update because of the params passed in the brackets